src/client.cpp

```diff
@@ -626,12 +618,16 @@ QString CClient::SetSndCrdDev ( const QString strNewDev )
         Sound.Start();
     }
 
-    // in case of an error inform the GUI about it
     if ( !strError.isEmpty() )
     {
-        emit SoundDeviceChanged ( strError );
+        // due to error, disconnect
+        Disconnect();
     }
 
+    // in case of an error, this will inform the GUI about it
+
+    emit SoundDeviceChanged();
```

Comment:
Why _not_ pass the error?  What's that got to do with this PR?

Reply:
Nothing, I think.